pkg/controller/starterkit/starterkit_controller.go

```diff
@@ -418,9 +436,82 @@ func (r *ReconcileStarterKit) Reconcile(request reconcile.Request) (reconcile.Re
 		reqLogger.Info("Skip reconcile: Deployment already exists", "Deployment.Namespace", foundDeployment.Namespace, "Deployment.Name", foundDeployment.Name)
 	}
 
+	// Check if the StarterKit instance is marked to be deleted, which is
+	// indicated by the deletion timestamp being set.
+	isMarkedToBeDeleted := instance.GetDeletionTimestamp() != nil
+	if isMarkedToBeDeleted {
+		if contains(instance.GetFinalizers(), starterkitFinalizer) {
+			// Run finalization logic for starterkitFinalizer. If the
+			// finalization logic fails, don't remove the finalizer so
+			// that we can retry during the next reconciliation.
+			if err := r.finalizeStarterKit(reqLogger, request, instance, client); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			// Remove starterkitFinalizer. Once all finalizers have been
+			// removed, the object will be deleted.
+			controllerutil.RemoveFinalizer(instance, starterkitFinalizer)
+			err := r.client.Update(ctx, instance)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		return reconcile.Result{}, nil
+	}
+
+	// Add finalizer for this CR
+	if !contains(instance.GetFinalizers(), starterkitFinalizer) {
+		reqLogger.Info("Adding finalizer to starterkit")
```

Comment:
There are a few different spelling/capitalizations for "starter kit" in the logs. Could you make that more consistent?